### PR TITLE
593 update workflow endpoints

### LINKED ISSUE #593: `berkeley`: Update activity related endpoints
**Is your feature request related to a problem? Please describe.**
There are a number of endpoints for workflows with activity in the name, Activity doesn't exist as a class anymore in the berkeley schema, its been updated to PlannedProcess with WorkflowExecution being the child class that best maps to what was previously grouped under activities. The following endpoints should be renamed with descriptions and parameters also updated to remove references to activity
- [x] https://api-berkeley.microbiomedata.org/docs#/workflows/post_activity_workflows_activities_post suggested replacement endpoint name is `workflow_execution_post`
- [x] https://api-berkeley.microbiomedata.org/docs#/find/find_activities_activities_get suggested replacement endpoint name is `find_workflow_execution_get`
- [x] https://api-berkeley.microbiomedata.org/docs#/find/find_activity_by_id_activities__activity_id__get. suggested replacement endpoint name is `find_workflow_execution_by_id_get`

- [x] remove https://api-berkeley.microbiomedata.org/docs#/v1/post_activity_v1_workflows_activities_post, it has been deprecated for a while
- [x] remove https://api-berkeley.microbiomedata.org/docs#/workflow_execution_activities/post_activity_v1_workflows_activities_post, it has been deprecated for a while

**Describe the solution you'd like**
In the [berkeley branch ](https://github.com/microbiomedata/nmdc-runtime/tree/berkeley)of nmdc_runtime remove language about activities from endpoint names, descriptions, filters.

**Acceptance Criteria**
- grep with activit on nmdc-repo returns on matches for endpoint names and filter fields. Text coming from https://github.com/microbiomedata/NMDC_documentation/blob/main/docs/howto_guides/api_gui.md will be updated separately after the endpoints themselves are updated.

Example scenario-based template:
Given I'm looking at [the swagger UI for the berkeley branch of nmdc-runtime](https://api-berkeley.microbiomedata.org/docs#/) , when I look a the endpoints, then there won't be any references to activity/activities.

**Describe alternatives you've considered**
Leave inconsistent and confusing endpoint names in place. 

cc @brynnz22 @mbthornton-lbl @picowatt @corilo @SamuelPurvine @shreddd

### PR DESCRIPTION
**Pull request by PeopleMakeCulture**
# Description
- removes all v1 endpoints, leaves a router with 410 error
- deprecates all /find/activities endpoints
- deprecates /workflows/activities endpoint
- creates /find/workflow_executions endpoints
- creates /workflows/workflow_executions endpoint

Fixes #593

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Tested locally via fastapi

#### TODO
- [ ] update `/workflows/workflow_executions` test
- [x] create `/find/workflow_executions` tests
- [x] fix ValueError in `pi` test object in `tests/test_data/test_gold_translator.py::test_get_pi`: https://github.com/microbiomedata/nmdc-runtime/pull/621/files

# Definition of Done (DoD) Checklist:
A simple checklist of necessary activities that add verifiable/demonstrable value to the product by asserting the quality of a feature, not the functionality of that feature; the latter should be stated as acceptance criteria in the issue this PR closes. Not all activities in the PR template will be applicable to each feature since the definition of done is intended to be a comprehensive checklist. Consciously decide the applicability of value-added activities on a feature-by-feature basis.

- [x] My code follows the style guidelines of this project (have you run `black nmdc_runtime/`?)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (in `docs/` and in <https://github.com/microbiomedata/NMDC_documentation/>?)
- [x] I have added tests that prove my fix is effective or that my feature works, incl. considering downstream usage (e.g. <https://github.com/microbiomedata/notebook_hackathons>) if applicable.
- [x] New and existing unit and functional tests pass locally with my changes (`make up-test && make test-run`)

### CODE REVIEW

**Comment by PeopleMakeCulture** [comment]
> I considered leaving the old endpoints and adding a 410, but found this confusing.
> TODO: If we can hide the old endpoints from the swaggerUI docs page, I am fine with leaving those endpoints up with a 410.

This can by done by setting `include_in_schema to False in the decorator. See: users.py:
```python
@router.get("/orcid_code", response_class=RedirectResponse, include_in_schema=False)
```

File: nmdc_runtime/api/endpoints/find.py [hunk 1]
```diff
@@ -176,44 +176,46 @@ def find_data_object_by_id(
...
     req: FindRequest = Depends(),
     mdb: MongoDatabase = Depends(get_mongo_db),
 ):
+    # TODO: Update description to be accurate wrt Berkeley schema
     """
-    The GET /activities endpoint is a general way to fetch metadata about various activities (e.g. metagenome assembly,
+    The GET /workflow_executions endpoint is a general way to fetch metadata about various workflow_executions (e.g. metagenome assembly,
     natural organic matter analysis, library preparation, etc.). Any "slot" (a.k.a. attribute) for
-    [WorkflowExecutionActivity](https://microbiomedata.github.io/nmdc-schema/WorkflowExecutionActivity/)
+    [WorkflowExecution](https://microbiomedata.github.io/nmdc-schema/WorkflowExecution/)
```
Comment (dwinston) [inline nmdc_runtime/api/endpoints/find.py:191]: Does not currently resolve. Ideally we'd take this opportunity to use our intended persistent ID here, i.e. <https://w3id.org/nmdc/WorkflowExecution>.

For now, use <https://microbiomedata.github.io/berkeley-schema-fy24/WorkflowExecution> and leave a note to refactor to use the `w3id.org` URL once that is active.

File: nmdc_runtime/api/endpoints/find.py [hunk 2]
```diff
@@ -176,44 +176,46 @@ def find_data_object_by_id(
...
     or [PlannedProcess](https://microbiomedata.github.io/nmdc-schema/PlannedProcess/) classes may be used in the filter
-    and sort parameters, including attributes of subclasses of *WorkflowExecutionActivity* and *PlannedProcess*.
+    and sort parameters, including attributes of subclasses of *WorkflowExecution* and *PlannedProcess*.
 
-    For example, attributes used in subclasses such as MetabolomicsAnalysisActivity (subclass of *WorkflowExecutionActivity*)
+    For example, attributes used in subclasses such as MetabolomicsAnalysis (subclass of *WorkflowExecution*)
     or [Extraction](https://microbiomedata.github.io/nmdc-schema/Extraction/) (subclass of *PlannedProcess*),
     can be used as input criteria for the filter and sort parameters of this endpoint.
     """
     return find_resources_spanning(req, mdb, activity_collection_names(mdb))
```
Comment (dwinston) [inline nmdc_runtime/api/endpoints/find.py:199]: Want to refactor name of `activity_collection_names` function as well, yes? `workflow_execution_collection_names`? Or I forget if there's only one collection in the new schema. Even if no change now, please leave comment about this concern here in the code.

File: nmdc_runtime/api/endpoints/workflows.py [hunk 3]
```diff
@@ -54,24 +53,29 @@ def list_workflow_capabilities(
...
-# TODO: Create activity.py in ../models
-@router.post("/workflows/activities")
-async def post_activity(
-    activity_set: dict[str, Any],
+@router.post("/workflows/activities", status_code=410)
+async def post_activity():
+    """
+    DEPRECATED: migrate all workflows from this endpoint to `/workflows/workflow_executions`.
+    """
+    return f"DEPRECATED: POST your request to `/workflows/workflow_executions` instead."
```
Comment (dwinston) [inline nmdc_runtime/api/endpoints/workflows.py:61]: need to annotate response string with `status_code`?

File: tests/test_api/test_endpoints.py [hunk 4]
```diff
@@ -236,12 +236,13 @@ def test_submit_changesheet():
...
+# THIS TEST IS SKIPPED - FOR NOW JUST UPDATE NAMING
 @pytest.mark.skip(
-    reason="Skipping because race condition causes  http://fastapi:8000/nmdcschema/ids/nmdc:wfrqc-11-t0tvnp52.2 to 404?"
+    reason="Skipping because race condition causes http://fastapi:8000/nmdcschema/ids/nmdc:wfrqc-11-t0tvnp52.2 to 404?"
 )
-def test_submit_workflow_activities(api_site_client):
+def test_submit_workflow_executions(api_site_client):
     test_collection, test_id = (
-        "read_qc_analysis_activity_set",
+        "read_qc_analysis_activity_set",  # does this exist in Berkeley schema?
```
Comment (dwinston) [inline tests/test_api/test_endpoints.py:245]: looks like no. From <https://microbiomedata.github.io/berkeley-schema-fy24/Database/>, I think there may only be [workflow_execution_set](https://microbiomedata.github.io/berkeley-schema-fy24/workflow_execution_set/).

**Review by dwinston — CHANGES_REQUESTED** [review]

File: nmdc_runtime/api/endpoints/find.py (hunk 2 above)
Comment (aclum) [reply nmdc_runtime/api/endpoints/find.py:199]: There is one collection for all WorkflowExecution documents in berkeley schema, [workflow_execution_set](https://microbiomedata.github.io/berkeley-schema-fy24/workflow_execution_set/)

File: tests/test_api/test_endpoints.py (hunk 4 above)
Comment (aclum) [reply tests/test_api/test_endpoints.py:245]: I agree with Donny. Also im curious why this test passes with this no longer existing collection. 
From this PR:
 tests/test_api/test_endpoints.py ss.....s.                               [ 25%]

File: nmdc_runtime/api/endpoints/find.py (hunk 1 above)
Comment (aclum) [reply nmdc_runtime/api/endpoints/find.py:191]: If updating to the berkeley fork please make a ticket/note/todo about updating to nmdc-schema ahead of the official rollout.

File: tests/test_api/test_endpoints.py (hunk 4 above)
Comment (aclum) [reply tests/test_api/test_endpoints.py:245]: oh nevermind, i see some other comments about this being skipped.

**Pull request merged**